docs/proposals/backup-restore.md

```diff
@@ -0,0 +1,111 @@
+# Backup and Restore Commands
```

Comment:
@TerryHowe Would you mind restructure this proposal doc based on the ORAS Enhancement Proposal Template? That would makes this proposal more informative and improves clarity https://github.com/oras-project/oras/blob/main/docs/proposals/proposal-doc-template.md